### DISCUSSION TITLE
Multiple PID instance

### DISCUSSION THREAD

**Discussion by krishy2** [arendst/Tasmota#18181] [Ideas]
How is it possible to have multiple PID instances, like thermostat? I think in ESP32 4 instances could fit.

**Comment by krishy2** [comment]
The PID driver is designed as it was only capable of running temperature control. However it should be a standalone module, like Shutter.
Pid<x> should be used for parameters and pv, like an analog input or any other value.

**Reply by barbudor** [reply]
As of today this module has no active maintainer
Feel free to jump in and provide a PR to improve the feature
